app/javascript/components/layer.jsx

```diff
@@ -48,165 +47,159 @@ export function Layer({ layer, isOwner, layerId, activeTool, activeColor }) {
+            fontSize: 25,
+          });
+          pathRef.current.fullySelected = true;
+        }
+        if (activeTool !== 'Pen' && activeTool !== 'Text' && scope.project.selectedItems.length === 0) {
+          path.strokeColor = 'black';
+          path.strokeWidth = 3;
+          path.dashArray = [10, 12];
+        } else if (scope.project.selectedItems.length === 0 || (activeTool === 'Text')) {
+          pathRef.current.strokeColor = activeColor;
+          pathRef.current.strokeWidth = activeTool === 'Pen' ? 3 : 1;
+        }
+      };
+
+      scope.view.onMouseDrag = (event) => {
+        if (activeTool === 'Pen') {
+          pathRef.current.add(event.point);
+        } else if (activeTool === 'Eraser') {
+          path.add(event.point);
+        } else if (activeTool === 'Select') {
+          p2 = new Paper.Point(event.point.x, event.point.y);
+          if (scope.project.selectedItems.length === 0) {
+            rect.set(p1, p2);
+            path.segments = [
+              new Paper.Segment(new Paper.Point(rect.x, rect.y)),
+              new Paper.Segment(new Paper.Point(rect.x + rect.width, rect.y)),
+              new Paper.Segment(new Paper.Point(rect.x + rect.width, rect.y + rect.height)),
+              new Paper.Segment(new Paper.Point(rect.x, rect.y + rect.height)),
+            ];
+          }
+        }
+      };
+
+      scope.view.onMouseUp = () => {
+        if (activeTool === 'Pen') {
+          pathRef.current.simplify(10);
+          setPathState(oldPaths => [...oldPaths, pathRef.current]);
+          transmitDiff(pathRef.current);
+          pathRef.current = null;
+        } else if (activeTool === 'Eraser') {
+          const newPathState = [];
+          for (p of pathState) {
+            if (path.intersects(p)) {
+              p.remove();
+            } else {
+              newPathState.push(p);
+            }
+            setPathState(newPathState);
+          }
+          path.remove();
+          path = null;
+        } else if (activeTool === 'Select') {
+          if (scope.project.selectedItems.length === 0) {
+            for (p of pathState) {
+              // check for path
+              if (p.segments && p.segments.every((segment) => path.contains(segment.point))) {
+                p.fullySelected = true;
+              } else if (p.content && p.isInside(rect)) { // check for text
+                p.fullySelected = true;
+              }
+            }
+            path.remove();
+            path = null;
+          } else {
+            const newPathState = [];
+            for (p of pathState) {
+              if (scope.project.selectedItems.find((selectedItem) => selectedItem.id === p.id)) {
+                p.translate(p2.subtract(p1));
+                p.fullySelected = false;
+              }
+              newPathState.push(p);
+            }
+            setPathState(newPathState);
+          }
+        }
+      };
+
+      scope.view.onKeyDown = (event) => {
```

Comment:
Am I missing something, shouldn't there be a `transmitDiff` called on something like `onKeyUp`?

Reply:
Yes there are a few places missing `transmitDiff` including text diffs, deletion diffs, selection and moving diffs, and eraser diffs. I only added `transmitDiff` where I did so we could try and tackle the simplest case first, the addition of a pen diff.